model/registry/cloudfoundry.yaml

```diff
@@ -0,0 +1,140 @@
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the application instance index for applications
+          deployed on the runtime or the vm id for CloudFoundry components.
+
+          Application instrumentation should use the value from environment
+          variable `CF_INSTANCE_INDEX`.
+
+          When system components are instrumented, values from the [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `instance.id` should be set to `spec.id`.
+        examples:
+          - '0'
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+      - id: app.name
+        type: string
+        brief: >
+          The name of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_name`. This is the same value
+          as reported by `cf apps`.
+        examples: ['my-app-name']
+      - id: app.id
+        type: string
+        brief: >
+          The guid of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_id`. This is the same value as
+          reported by `cf app <app-name> --guid`.
+        examples: ['218fc5a9-a5f1-4b54-aa05-46717d0ab26d']
+      - id: space.name
+        type: string
+        brief: >
+          The name of the CloudFoundry space the application is running in.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.space_name`. This is the same value as
+          reported by `cf spaces`.
+        examples: ['my-space-name']
+      - id: space.id
+        type: string
+        brief: >
+          The guid of the CloudFoundry space the application is running in.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.space_id`. This is the same value as
+          reported by `cf space <space-name> --guid`.
+        examples: ['218fc5a9-a5f1-4b54-aa05-46717d0ab26d']
+      - id: org.name
+        type: string
+        brief: >
+          The name of the CloudFoundry organization the app is running in.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.org_name`. This is the same value as
+          reported by `cf orgs`.
+        examples: ['my-org-name']
+      - id: org.id
+        type: string
+        brief: >
+          The guid of the CloudFoundry org the application is running in.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.org_id`. This is the same value as
+          reported by `cf org <org-name> --guid`.
+        examples: ['218fc5a9-a5f1-4b54-aa05-46717d0ab26d']
+      - id: source.type
```

Comment:
I'm a bit confused by the `source.type` attribute. Is it changing over the course of the application? Is it a fixed value known when starting up? All the attributes here are defined as resources, which mean, they should be readily available when the application/SDK is initializing. I saw in the discussion that you were not sure if this is supposed to be a log/metric vs resource attribute?

Reply:
For any running component, the source.type will be fixed. However, a CloudFoundry applications can consist of multiple processes, that will have different source.type fixed at their respective deployment type. CloudFoundry will report all those components under the same app.id. Picture a K8s pod consisting of several containers. 

In addition CloudFoundry will add logs from system components with different source.types. Examples are Gorouter access logs with source.type=RTR or staging logs with source.type=STG. See https://docs.cloudfoundry.org/devguide/deploy-apps/streaming-logs.html#format for more examples. In each case, the source.type remains constant during runtime for the emitting component. Note, that even if those components emit logs for different applications, e.g. the Gorouter, the source.type remains fixed for the particular component.

Reply:
`source.type` is used to differentiate different log signals. It may be populated by both applications and system components. Here's the Cloud Foundry docs about it: https://docs.cloudfoundry.org/devguide/deploy-apps/streaming-logs.html#format.

Just considering applications, the `source.type` attribute does change over the course of the application. When an application is being staged, the `source.type` is set to `STG/0`. Once an application has started up the `source.type` is set to `APP/...`.

In Cloud Foundry, "applications" are comprised of many "processes" (e.g. "web", "worker", etc,), which each have some number of instances. The `source.type` will be populated differently depending on the application/process/instance that the signal originates from.

There's also tasks, but maybe we can ignore that for now.

Reply:
(My bad @KarstenSchnitter I was typing at the same time and didn't see your message before I sent mine 😅)

Reply:
So if source type is changing, we have a problem here as resource attributes are suppose to be immutable, meaning a resource is created and populated on app startup and after that it does not change.

Reply:
The source type is fixed, once an application is started. It is just a closer description of the kind of resource. The source.type `STG/0` is not emitted by the application, but by the CloudFoundry runtime. Every event emitted by the application will be of the same fixed `source.type`. If there are multiple processes running, each will have its own `process.id` and possibly a custom `source.type`. However, once a process is started, the `source.type` is fixed just like the `process.id`.